src/docs/cookie-clicker-swiftui.md

```diff
@@ -0,0 +1,314 @@
+---
+title: Cookie Clicker with SwiftUI
+---
+
+This tutorial is an introduction to the world of iOS development using Apple's latest UI framework: [SwiftUI](https://developer.apple.com/xcode/swiftui/)
+
+We'll be building a counter app, where the user will press a button and increment the number on the screen.
+
+## Pre-requisites
+
+In order to work through this tutorial, you will need a Mac with Xcode 11 installed. Xcode 11 is the minimum version that supports SwiftUI. It is useful to have Mac OSX Catalina installed to see live previews in Xcode, but you can still use Mac OS X Mojave.
+
+Download and install Xcode from the App Store. This may take a while — its a big program. If you are at a Codebar ask the coaches before downloading it from the App Store as they frequently have a copy on a USB drive which will save some installation time, but it would be better if you come to the workshop with it installed.
+
+## Creating a Project
+
+### 1. Open Xcode and click Create a new Xcode project
+
+![step1](assets/cookie_clicker_swiftui/step1.png)
+
+### 2. Select Single View Application from the **iOS > Application list**, and click **Next**
+
+![step2](assets/cookie_clicker_swiftui/step2.png)
+
+### 3. Fill in the project details:
+
+- Product Name: Clicker
+- Team: None
+- Organization Name: Whatever you want — your name is always a good fill-in
+- Organization Identifier: com.(OrganizationName)
+
+Make sure Swift is selected as the language.
+  
+Make sure SwiftUI is selected as the User Interface.
+
+Untick the 3 boxes at the bottom.
+
+Click Next
+
+![step3](assets/cookie_clicker_swiftui/step3.png)
+
+### 4. Select a sensible place to save your project, then hit **Create**.
+
+Take some time to look around the project that's been created with your coach. What files are there? How do you run your app?
+  
+If you are running Mac OS X Catalina, try running the preview to see what the default project provides.
+
+## Add a Button
+
+### 5. Open up the **Navigation Area** in Xcode, and then the `ContentView.swift` file and its associated Canvas.
+
+![step4](assets/cookie_clicker_swiftui/step4.gif)
+  
+If you click on the `Resume` button in the top right, and you are running Mac OS X Catalina you will see a preview of the **Hello World** application that the template has provided for you.
```

Comment:
this took a few minutes for my macbook to load the first time with 8GB ram, is this something that might take some time for students to see too?

Reply:
Yes. Usually it's because you've just installed or upgraded Xcode, or because you've chosen a simulator that you've never used before. Their is some initialisation to the simulator the first time. I know you're running in preview, but it sort of uses a simulator. Subsequent loads are much faster.

Reply:
great - i'm happy to know it's not just me 😆 ! We can let students know to expect this may take some time :)